### PR TITLE
add libtai-mux

### PR DESCRIPTION
**Pull request by ishidawataru**
Signed-off-by: Wataru Ishida <ishida@nel-america.com>

### CODE REVIEW

**Comment by ishidawataru** [comment]
@dileepbk @scott-emery prototyped a TAI library ( `libtai-mux.so` ) to support multiple types of optical modules based on our discussion. Please take a look and give me your comments.

File: tai_mux/module_adapter.hpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+                return TAI_STATUS_FAILURE;
+            }
+            return m_module_api->set_module_attributes(module_id, attr_count, attr_list);
+        }
+
+        tai_status_t get_module_attributes(
+            _In_ tai_object_id_t module_id,
+            _In_ uint32_t attr_count,
+            _In_ tai_attribute_t *attr_list) {
+            if ( m_module_api == nullptr && m_module_api->get_module_attributes == nullptr ) {
```
Comment (scott-emery) [inline tai_mux/module_adapter.hpp]: I think these tests should be 'OR' (||), not 'AND' (&&).

File: tai_mux/static_platform_adapter.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#include "static_platform_adapter.hpp"
+
+StaticPlatformAdapter::StaticPlatformAdapter(uint64_t flags, const tai_service_method_table_t* services) {
+    auto config_file = std::string(std::getenv(TAI_MUX_STATIC_CONFIG_FILE.c_str()));
```
Comment (scott-emery) [inline tai_mux/static_platform_adapter.cpp]: You might want to consider having a default location for the config file which is used in the case when the environment variable is not set.

**Review by scott-emery — APPROVED** [review]
Just two minor comments from me. Other than those this looks really good. Thanks for submitting this!

File: tai_mux/static_platform_adapter.cpp (hunk 2 above)
Comment (ishidawataru) [reply tai_mux/static_platform_adapter.cpp]: Do you think `/etc/tai/mux/static.json` is a reasonable place for it?

Comment (scott-emery) [reply tai_mux/static_platform_adapter.cpp]: Yes, that sounds like a good default location.

**Comment by ishidawataru** [comment]
@scott-emery thank you for reviewing. reflected your comments. PTAL.

**Comment by ishidawataru** [comment]
@dileepbk Thanks for your review! Since this PR is already merged, I'll address your comments in a separate PR.

**Pull request merged**